lightning/src/chain/channelmonitor.rs

```diff
@@ -1571,7 +1571,7 @@ impl<Signer: Sign> ChannelMonitorImpl<Signer> {
 			// First, process non-htlc outputs (to_holder & to_counterparty)
 			for (idx, outp) in tx.output.iter().enumerate() {
 				if outp.script_pubkey == revokeable_p2wsh {
-					let revk_outp = RevokedOutput::build(per_commitment_point, per_commitment_key, self.counterparty_tx_cache.counterparty_delayed_payment_base_key, self.counterparty_tx_cache.counterparty_htlc_base_key, InputDescriptors::RevokedOutput, outp.value, None, self.counterparty_tx_cache.on_counterparty_tx_csv);
+					let revk_outp = RevokedOutput::build(per_commitment_point, per_commitment_key, self.counterparty_tx_cache.counterparty_delayed_payment_base_key, self.counterparty_tx_cache.counterparty_htlc_base_key, WEIGHT_REVOKED_OUTPUT, outp.value, None, self.counterparty_tx_cache.on_counterparty_tx_csv);
```

Comment:
This feels like even more of a layer violation than the original `InputDescriptors` enum. I think my original complaint about it was more about how its confusing to have two separate enums (and that you can have a non-`Revoked` `InputDescriptor` inside of a `RevokedOutput` which is obviously nonsensical). Instead, can we make `RevokedOutput` at least contain an enum so that the `htlc` field isn't an option and we don't need to store a weight passed in from another layer and can calculate it locally?

Reply:
Addressed with new RevokedHTLCOutput and corresponding constructor